tests/integration/dynamic_ip_test.yaml

```diff
@@ -0,0 +1,1467 @@
+interfaces:
+- name: dummy00
+  type: dummy
+  state: up
+  ipv4:
+    enabled: true
+    dhcp: true
+    auto-dns: true
+    auto-gateway: true
+    auto-routes: true
+    auto-route-table-id: 0
+  ipv6:
+    enabled: true
+    dhcp: true
+    autoconf: false
+    auto-dns: true
+    auto-gateway: true
+    auto-routes: true
+    auto-route-table-id: 0
+description: Bring up the dummy device dummy00
+interfaces:
+- name: dummy00
+  type: dummy
+  state: up
+description: Configure the dhcpcli ethernet interface with DHCP4, IPv4 auto dns, IPv4 auto gateway, IPv4 auto routes, IPv4 auto route table ID 0, DHCP6, autoconf, IPv6 auto dns, IPv6 auto gateway, IPv6 auto routes, IPv6 auto route table ID 0
+interfaces:
+- name: dhcpcli
+  type: ethernet
+  state: up
+  ipv4:
+    enabled: true
+    dhcp: true
+    auto-dns: true
+    auto-gateway: true
+    auto-routes: true
+    auto-route-table-id: 0
+  ipv6:
+    enabled: true
+    dhcp: true
+    autoconf: true
+    auto-dns: true
+    auto-gateway: true
+    auto-routes: true
+    auto-route-table-id: 0
+description: configure the ethernet device dhcpcli with IPv6 disabled
+interfaces:
+- name: dhcpcli
+  type: ethernet
+  state: up
+  ipv6:
+    enabled: false
+description: Remove the interface dhcpcli
+interfaces:
+- name: dhcpcli
+  type: ethernet
+  state: absent
+description: Configure the ethernet interface dhcpcli with DHCP4, IPv4 auto dns, IPv4 auto gateway, IPv4 auto routes, IPv4 auto route table ID 0
+interfaces:
+- name: dhcpcli
+  type: ethernet
+  state: up
+  ipv4:
+    enabled: true
+    dhcp: true
+    auto-dns: true
+    auto-gateway: true
+    auto-routes: true
+    auto-route-table-id: 0
+description: 'Configure the ethernet interface dhcpcli with IPv4 dhcp enabled, do not apply IPv4 DNS resolver information retrieved from DHCP server, IPv4 auto gateway, IPv4 auto routes, IPv4 auto route '
+interfaces:
+- name: dhcpcli
+  type: ethernet
+  state: up
+  ipv4:
+    enabled: true
+    dhcp: true
+    auto-dns: false
+    auto-gateway: true
+    auto-routes: true
+    auto-route-table-id: 0
+description: Configure the ethernet interface dhcpcli with IPv4 dhcp enabled, IPv4 auto dns, IPv4 auto routes, IPv4 auto route table ID 0, do not apply IPv4 default gateway retrieved from DHCP server
+interfaces:
+- name: dhcpcli
+  type: ethernet
+  state: up
+  ipv4:
+    enabled: true
+    dhcp: true
+    auto-dns: true
+    auto-gateway: false
+    auto-routes: true
+    auto-route-table-id: 0
+description: Configure the ethernet interface dhcpcli with DHCP4, IPv4 auto dns, IPv4 auto gateway, do not set IPv4 routes retrieved from DHCP server, IPv4 auto table ID 0
+interfaces:
+- name: dhcpcli
+  type: ethernet
+  state: up
+  ipv4:
+    enabled: true
+    dhcp: true
+    auto-dns: true
+    auto-gateway: true
+    auto-routes: false
+    auto-route-table-id: 0
+description: Configure the ethernet device dhcpcli, set the address 192.0.2.252/24
```

Comment:
Configure the ethernet device dhcpcli with static ip 192.0.2.252/24 and DHCPv4 disabled.